consensus/types/peer_round_state.go

```diff
@@ -55,3 +55,26 @@ func (prs PeerRoundState) StringIndented(indent string) string {
 		indent, prs.CatchupCommit, prs.CatchupCommitRound,
 		indent)
 }
+
+
+// Protobuf Compatiablity
+func (ps *PeerRoundState) Unmarshal(bs []byte) error {
+	return cdc.UnmarshalBinaryBare(bs, ps)
+}
+
+func (ps *PeerRoundState) Marshal() ([]byte, error) {
+	return cdc.MarshalBinaryBare(ps)
+}
+
+func (ps *PeerRoundState) MarshalTo(data []byte) (int, error) {
```

Comment:
Is the idea that `Size()` is called before calling `MarshalTo` to ensure the `data` buffer is correctly sized?

Reply:
@ebuchman, yes its used to check the size/len of the buffer.